x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/DlsFlsRequestCacheDifferentiator.java

```diff
@@ -49,22 +49,21 @@ public void accept(ShardSearchRequest request, StreamOutput out) throws IOExcept
             .getTransient(AuthorizationServiceField.INDICES_PERMISSIONS_KEY);
         final String indexName = request.shardId().getIndexName();
         IndicesAccessControl.IndexAccessControl indexAccessControl = indicesAccessControl.getIndexPermissions(indexName);
-        if (indexAccessControl != null) {
-            final boolean flsEnabled = indexAccessControl.getFieldPermissions().hasFieldLevelSecurity();
-            final boolean dlsEnabled = indexAccessControl.getDocumentPermissions().hasDocumentLevelPermissions();
-            if ((flsEnabled || dlsEnabled) && DOCUMENT_LEVEL_SECURITY_FEATURE.checkWithoutTracking(licenseState)) {
-                logger.debug(
-                    "index [{}] with field level access controls [{}] "
-                        + "document level access controls [{}]. Differentiating request cache key",
-                    indexName,
-                    flsEnabled,
-                    dlsEnabled
-                );
-                indexAccessControl.buildCacheKey(
-                    out,
-                    SecurityQueryTemplateEvaluator.wrap(securityContext.getUser(), scriptServiceReference.get())
-                );
-            }
+        if (indexAccessControl != null
```

Comment:
I think we should assert (or safeguardf) indexAccessControl cannot be null here.

Reply:
I'm not sure I should do that in this PR. I think we discussed this on our sync, it's still _theoretically_ possible that the `indexAccessControl` be `null`, whereas this PR is to remove the `#isGranted` field from non-null access control objects where it already was assumed to be `true`.